crates/services/txpool/src/ports.rs

```diff
@@ -36,12 +37,17 @@ use std::{
     sync::Arc,
 };
 
+#[async_trait::async_trait]
 pub trait PeerToPeer: Send + Sync {
     type GossipedTransaction: NetworkData<Transaction>;
 
     // Gossip broadcast a transaction inserted via API.
     fn broadcast_transaction(&self, transaction: Arc<Transaction>) -> anyhow::Result<()>;
 
+    /// Creates a stream that is filled with the peer_id when they subscribe to
+    /// our transactions gossip.
+    fn new_tx_subscription(&self) -> BoxStream<Vec<u8>>;
```

Comment:
should we use a PeerId newtype for Vec<u8>?

Reply:
I took example from :
https://github.com/FuelLabs/fuel-core/blob/95b373e80094b09ab5dc6c1e351deccf68ff7dd9/crates/services/p2p/src/service.rs#L842

I was thinking that this was the way you used `PeerId` outside of the P2P module. But I just saw that the PeerId type is on the type shared crate I will change this.

Reply:
Done